packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -162,7 +225,9 @@ function buildParseFragmentFn(
                 cacheKey |= FragmentCache.SHADOW_MODE_SYNTHETIC;
             }
 
-            if (!isUndefined(cache[cacheKey])) {
+            // Cache is only here to prevent calling innerHTML multiple times which doesn't happen on the server.
+            if (process.env.IS_BROWSER && !isUndefined(cache[cacheKey])) {
+                // Disable this on the server to prevent cache poisoning when expressions are used.
                 return cache[cacheKey];
             }
```

Comment:
```suggestion
            if (process.env.IS_BROWSER) {
                // Disable this on the server to prevent cache poisoning when expressions are used.
                const cached = cache[cacheKey];
                if (!isUndefined(cached)) {
                    return cached;
                }
            }
```

Avoid doing the object lookup twice in the browser. (Minor perf optimization.)

Reply:
[Updated](https://github.com/salesforce/lwc/pull/4056/commits/93f51d8c0d62013ab61fb6f499ce79329df14da1#diff-9d36a520650dfa1dd13874ab3870b6f688303b9e5860066178098aadcf45a860L229-L232)!